src/site/views/ticketpurchase/tmpl/summary.php

```diff
@@ -201,52 +237,172 @@ class="swa-option-selector"
+			fontSmoothing: "antialiased",
+			fontSize: "16px",
+			"::placeholder": {
+				color: "#32325d"
+			}
+		},
+		invalid: {
+			fontFamily: 'Arial, sans-serif',
+			color: "#fa755a",
+			iconColor: "#fa755a"
+		}
+	};
+	var card = elements.create("card", {
+		style: style
+	});
+	// Stripe injects an iframe into the DOM
+	card.mount("#card-element");
+	card.on("change", function(event) {
+		// Disable the Pay button if there are no card details in the Element or add on not valid
+		$stripeCardErrorMsg = event.error ? event.error.message : "";
+		$cardDetailsComplete = event.complete;
+		setStripeButtonStatus();
+	});
+	var form = document.getElementById("payment-form");
+
+	var payWithCard = function(stripe, card, clientSecret) {
+		// console.log(stripe, card, clientSecret)
+		loading(true);
+		stripe
+			.confirmCardPayment(clientSecret, {
+				payment_method: {
+					card: card
+				}
+			})
+			.then(function(result) {
+				console.log(result)
+				if (result.error) {
+					// Show error to your customer
+					showError(result.error.message);
+				} else {
+					// The payment succeeded!
+					processOrder(result.paymentIntent.id);
+				}
+			})
+			.catch(function(error) {
+				// Handle the error
+				console.error("Error here");
+			});
+	};
+	/* ------- UI helpers ------- */
+	// Shows a success message when the payment is complete
+	var processOrder = function(paymentIntentId) {
+		loading(true);
+		document.querySelector(".result-message").classList.remove("hidden");
+		document.querySelector("#stripe-button").disabled = true;
+		fetch("<?php echo JRoute::_('index.php??option=com_swa&task=ticketpurchase.addTicketToDb') ?>", {
+				method: "POST",
+				headers: {
+					"Content-Type": "application/json"
+				},
+				body: JSON.stringify({
+					paymentIntentId: paymentIntentId,
+				})
+			})
+			.then(function(result) {
+				if (result.ok) {
+					window.location.href = "<?php echo JRoute::_('index.php?option=com_swa&view=membertickets') ?>";
+				} else {
+					return result.text().then(text => {
+						throw new Error(text)
+					})
+				}
+			})
+
+	};
+	// Show the customer the error from Stripe if their card fails to charge
+	var showError = function(errorMsgText) {
+		loading(false);
+		var errorMsg = document.querySelector("#card-error");
+		errorMsg.textContent = errorMsgText;
+		setTimeout(function() {
+			errorMsg.textContent = "";
+		}, 7000);
+	};
+	// Show a spinner on payment submission
+	var loading = function(isLoading) {
+		if (isLoading) {
+			// Disable the button and show a spinner
+			document.querySelector("#stripe-button").disabled = true;
+			document.querySelector("#spinner").classList.remove("hidden");
+			document.querySelector("#button-text").classList.add("hidden");
+		} else {
+			document.querySelector("#stripe-button").disabled = false;
+			document.querySelector("#spinner").classList.add("hidden");
+			document.querySelector("#button-text").classList.remove("hidden");
+		}
+	};
```

Comment:
The spinner only starts spinning after the paymentIntent is requested and received, which could take some time on slow internets. Can this easily be started as soon as the button is pressed? No worries if not easy to do.